agent/stats/queue_test.go

```diff
@@ -309,3 +310,55 @@ func TestCpuStatsSetNotSetToInfinity(t *testing.T) {
 		t.Errorf("Computed cpuStatsSet.SampleCount (%d) != expected value (%d)", sampleCount, 1)
 	}
 }
+
+func TestResetThresholdElapsed(t *testing.T) {
+	// create a queue
+	queueLength := 3
+	queue := NewQueue(queueLength)
+
+	queue.Reset()
+
+	thresholdElapsed := queue.resetThresholdElapsed(2 * time.Millisecond)
+	assert.False(t, thresholdElapsed, "Queue is NOT expected to elapsed the threshold right after reset")
+
+	time.Sleep(3 * time.Millisecond)
+	thresholdElapsed = queue.resetThresholdElapsed(2 * time.Millisecond)
+
+	assert.True(t, thresholdElapsed, "Queue is expected to elapse the threshold")
```

Comment:
Same as above. I think the better message here is "Queue reset threshold is expected to elapse after waiting"